Complete refactoring of recipe configurations.

godofredoc opened the pull request:
Bug:
  https://github.com/flutter/flutter/issues/56419

christopherfujino commented on config/iostools_config.star:
```diff
@@ -0,0 +1,58 @@
+"""
+Configurations for ios_tool builders.
+
+The schedulers pull commits indirectly from GoB repo (https://chromium.googlesource.com/external/github.com/flutter/flutter)
```
commits actually come from https://flutter-mirrors.googlesource.com/

godofredoc replied:
Updated.

christopherfujino commented on config/cocoon_config.star:
```diff
@@ -0,0 +1,51 @@
+"""
+Configurations for the cocoon repository.
+
+The schedulers pull commits indirectly from GoB repo (https://chromium.googlesource.com/external/github.com/flutter/flutter)
```
pulls from https://chromium.googlesource.com/external/github.com/flutter/cocoon

godofredoc replied:
Updated

christopherfujino commented on config/packaging_config.star:
```diff
@@ -0,0 +1,114 @@
+
+    # Defines console views for prod builders
+    console_view_name = ('packaging'
+                         if branch == 'master' else '%s_packaging' % branch)
```
this function should never get called with `branch == 'master'`, right?

christopherfujino replied:
Should we throw here (or somehow escalate a warning in a way we'll see) if this function ever gets invoked on the master branch? I'm worried a future refactor will lead to us building packages for every single master commit. At the minimum, we should exit early.

godofredoc replied:
Added an explanation and added functionality to fail if we ever try to build from master.

godofredoc commented:
> 2 nits about comments and then a question about ensuring we don't regress in the future and build packages on master.

Updated the PR addressing the comments.
